src/midiEditorCommands.cpp

```diff
@@ -184,13 +183,8 @@ void previewNotes(MediaItem_Take* take, const vector<MidiNote>& notes) {
 		previewReg.src = &previewSource;
 		previewReg.m_out_chan = -1; // Use .preview_track.
 	}
-	if (previewDoneTimer) {
-		// Notes are currently being previewed. Interrupt them.
-		// We want to turn off these notes immediately.
-		KillTimer(NULL, previewDoneTimer);
-		previewDoneTimer = 0;
-		previewNotesOff();
-	}
+	// Stop the current preview.
+	cancelMidiPreviewDoneTimer(true);
 	// Queue note on events for the new notes.
```

Comment:
The reason I didn't do this originally is that there's no guarantee concerning the timing or sync/async behaviour of PlayTrackPreview. In the original code, we only call PlayTrackPreview once for both note offs and note ons here. In your new code, PlayTrackPreview will be called twice in quick succession. I worry that might cause problems if PlayTrackPreview is async. I guess you could get around this by having another argument to the cancel function to only queue.